pkg/odo/cli/component/push.go

```diff
@@ -63,13 +69,41 @@ func (po *PushOptions) Complete(name string, cmd *cobra.Command, args []string)
 
 	// if experimental mode is enabled and devfile is present
 	if experimental.IsExperimentalModeEnabled() && util.CheckPathExists(po.DevfilePath) {
-		envinfo, err := envinfo.NewEnvSpecificInfo(po.componentContext)
+		envInfo, err := envinfo.NewEnvSpecificInfo(po.componentContext)
 		if err != nil {
 			return errors.Wrap(err, "unable to retrieve configuration information")
 		}
-		po.EnvSpecificInfo = envinfo
+		po.EnvSpecificInfo = envInfo
 		po.Context = genericclioptions.NewDevfileContext(cmd)
 
+		devfileCommandChanged := false
+		var pushCommands string
+		pushCommand := po.EnvSpecificInfo.GetPushCommand()
+		if po.devfileInitCommand != "" {
+			pushCommand.Init = strings.ToLower(po.devfileInitCommand)
+			pushCommands = fmt.Sprintf("%s Init: %s", pushCommands, pushCommand.Init)
+			devfileCommandChanged = true
+		}
+		if po.devfileBuildCommand != "" {
+			pushCommand.Build = strings.ToLower(po.devfileBuildCommand)
+			pushCommands = fmt.Sprintf("%s Build: %s", pushCommands, pushCommand.Build)
+			devfileCommandChanged = true
+		}
+		if po.devfileRunCommand != "" {
+			pushCommand.Run = strings.ToLower(po.devfileRunCommand)
+			pushCommands = fmt.Sprintf("%s Run: %s", pushCommands, pushCommand.Run)
+			devfileCommandChanged = true
+		}
+
+		// only if the devfile command changed, update env.yaml and display the information
+		if devfileCommandChanged {
+			err = po.EnvSpecificInfo.SetConfiguration("push", pushCommand)
+			if err != nil {
+				return errors.Wrap(err, "unable to set push configuration for env.yaml")
+			}
+			log.Italicf("The default devfile command(s) has changed for%s", pushCommands)
```

Comment:
I don't think we need the (s) as we the pointing the exact command

Reply:
updated